classes/Tools.php

```diff
@@ -1698,23 +1698,23 @@ public static function truncateString($text, $length = 120, $options = [])
                 return $text;
             }
 
-            $truncate = Tools::substr($text, 0, $length - Tools::strlen($ellipsis));
+            $truncate = Tools::substr($text, 0, $length - Tools::strlen($ellipsis ?? ''));
         }
 
-        if (!$exact) {
-            $spacepos = Tools::strrpos($truncate, ' ');
-            if ($html) {
-                $truncate_check = Tools::substr($truncate, 0, $spacepos);
+        if (!isset($exact) || !$exact) {
```

Comment:
why not reassign $truncate once

$truncate = $truncate ?? '';

and avoid repeating the check